### PR TITLE
Remove dataframes crate and feature

### PR DESCRIPTION
**Pull request by IanManske**
# Description
Removes the old `nu-cmd-dataframe` crate in favor of the polars plugin. As such, this PR also removes the `dataframe` feature, related CI, and full releases of nushell.

### CODE REVIEW

File: .github/workflows/nightly-build.yml [hunk 1]
```diff
@@ -84,41 +84,30 @@ jobs:
         include:
         - target: aarch64-apple-darwin
           os: macos-latest
-          target_rustflags: ''
```
Comment (fdncred) [inline .github/workflows/nightly-build.yml:87]: I'm pretty sure that not having `target_rustflags: ''` causes winget to fail. there's a closed issue on it in the repo somewhere. don't ask me to explain it.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -29,17 +29,6 @@ jobs:
         # instead of 14 GB) which is too little for us right now. Revisit when `dfr` commands are
         # removed and we're only building the `polars` plugin instead
         platform: [windows-latest, macos-13, ubuntu-20.04]
-        feature: [default, dataframe]
-        include:
-          - feature: default
-            flags: ""
```
Comment (fdncred) [inline .github/workflows/ci.yml:35]: same comment about flags as below. i'll try to find the issue about winget and flags to make sure if it's this flags matters or not

File: .github/workflows/release-pkg.nu [hunk 3]
```diff
@@ -9,19 +9,16 @@
 # Instructions for manually creating an MSI for Winget Releases when they fail
 # Added 2022-11-29 when Windows packaging wouldn't work
 # Updated again on 2023-02-23 because msis are still failing validation
-# Update on 2023-10-18 to use RELEASE_TYPE env var to determine if full or not
 # To run this manual for windows here are the steps I take
 # checkout the release you want to publish
 # 1. git checkout 0.86.0
 # unset CARGO_TARGET_DIR if set (I have to do this in the parent shell to get it to work)
-# 2. $env:CARGO_TARGET_DIR = ""
+# 2. $env.CARGO_TARGET_DIR = ""
```
Comment (fdncred) [inline .github/workflows/release-pkg.nu:16]: This was actually correct syntax for powershell which is the shell I use as the parent shell mentioned above this line.

**Comment by fdncred** [comment]
I started making comments but instead of going through the changes, unrelated to rust code and specific to winget, we need to have the rust flags in order for winget to work. I couldn't remember which flags need to be maintained so I went to lookup the issues which have them and describe why they're needed. Please restore these if/where they're missing.

https://github.com/nushell/nushell/pull/9513
https://github.com/nushell/nushell/pull/9322#issuecomment-1602932971

File: .github/workflows/release-pkg.nu [hunk 4]
```diff
@@ -9,19 +9,16 @@
...
 # 2. hide-env CARGO_TARGET_DIR
 # 3. $env.TARGET = 'x86_64-pc-windows-msvc'
-# 4. $env.TARGET_RUSTFLAGS = ''
```
Comment (fdncred) [inline .github/workflows/release-pkg.nu:20]: Restore here too

File: .github/workflows/release-pkg.nu [hunk 5]
```diff
@@ -48,31 +45,15 @@ let os = $env.OS
 let target = $env.TARGET
 # Repo source dir like `/home/runner/work/nushell/nushell`
 let src = $env.GITHUB_WORKSPACE
-let flags = $env.TARGET_RUSTFLAGS
```
Comment (fdncred) [inline .github/workflows/release-pkg.nu:51]: and here, please restore

File: .github/workflows/release-pkg.nu [hunk 6]
```diff
@@ -91,23 +72,23 @@ if $os in ['macos-latest'] or $USE_UBUNTU {
         'aarch64-unknown-linux-gnu' => {
             sudo apt-get install gcc-aarch64-linux-gnu -y
             $env.CARGO_TARGET_AARCH64_UNKNOWN_LINUX_GNU_LINKER = 'aarch64-linux-gnu-gcc'
-            cargo-build-nu $flags
+            cargo-build-nu
```
Comment (fdncred) [inline .github/workflows/release-pkg.nu:75]: I believe all these places need the $flag restored too

File: .github/workflows/release.yml [hunk 7]
```diff
@@ -34,41 +34,30 @@ jobs:
         include:
         - target: aarch64-apple-darwin
           os: macos-latest
-          target_rustflags: ''
```
Comment (fdncred) [inline .github/workflows/release.yml:37]: restore these please

File: .github/workflows/release-pkg.nu (hunk 3 above)
Comment (IanManske) [reply .github/workflows/release-pkg.nu:16]: Oh I see, so that line is in the parent shell, and the rest are in nushell?

Comment (fdncred) [reply .github/workflows/release-pkg.nu:16]: yup, my comments are pretty terrible here. sry.

**Comment by IanManske** [comment]
Oh ok, I see the issue with the `RUSTFLAGS`. However, setting `target_rustflags: ''` in CI means that the `RUSTFLAGS="-D warnings"` set by the `setup-rust-toolchain` action is overwritten. I believe this means that warnings will be ignored (e.g., unused imports like [here](https://github.com/nushell/nushell/pull/12883/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fc)). For building releases I think it makes sense to restore the old flags.

File: .github/workflows/release-pkg.nu (hunk 3 above)
Comment (IanManske) [reply .github/workflows/release-pkg.nu]: all good lol

**Comment by fdncred** [comment]
> Oh ok, I see the issue with the `RUSTFLAGS`. However, setting `with: rustflags: ""` in CI means that the `RUSTFLAGS="-D warnings"` set by the `setup-rust-toolchain` action is overwritten. I believe this means that warnings will be ignored (e.g., unused imports like [here](https://github.com/nushell/nushell/pull/12883/files#diff-42cb6807ad74b3e201c5a7ca98b911c5fa08380e942be6e4ac5807f8377f87fc)). For building releases I think it makes sense to restore the old flags.

Hopefully winget will pass with these changes. Otherwise, I'll be pinging you to reverse the ci changes.

**Comment by IanManske** [comment]
Actually what I have right now won't work because of #12900. I'll restore the old flag code for the release workflows @fdncred.

**Comment by fdncred** [comment]
I think you're one the right track with removing full, I was just worried about the rust flags parts.

**Comment by IanManske** [comment]
> I was just worried about the rust flags parts.

With good reason XD

The `target_rustflags` and `flags` were for feature flags (e.g., `--features=dataframes`), so I kept those removed. (`$env.TARGET_RUSTFLAGS` is not used/recognized by cargo.)

I also kept the `rustflags`/`RUSTFLAGS` as removed in the CI for the reason mentioned before.

Otherwise, feel free to double check that the release workflows have the same `rustflags`/`RUSTFLAGS` as before.

**Comment by fdncred** [comment]
It's pretty clear that all the workflows are different. Only time will tell as to whether winget-pkgs will accept our next release.

**Review by sholderbach — APPROVED** [review]
What I can see by looking at the workflows etc. looks all plausible to me

File: crates/nu_plugin_polars/Cargo.toml [hunk 8]
```diff
@@ -4,7 +4,7 @@ description = "Nushell dataframe plugin commands based on polars."
 edition = "2021"
 license = "MIT"
 name = "nu_plugin_polars"
-repository = "https://github.com/nushell/nushell/tree/main/crates/nu-cmd-dataframe"
+repository = "https://github.com/nushell/nushell/tree/main/crates/nu_plugin_polars"
```
Comment (sholderbach) [inline crates/nu_plugin_polars/Cargo.toml:7]: Good catch!

**Comment by fdncred** [comment]
if we're going to land this, we should do it soon enough to be able to use it a bit before the release.

**Pull request merged**